From 6a95958815eba2767d68bc31b01afb4b9037524d Mon Sep 17 00:00:00 2001
From: Henrik Grimler <grimler@termux.dev>
Date: Wed, 26 Mar 2025 17:52:26 +0100
Subject: [PATCH] termux-api: exec_am_broadcast: only memcpy argc-2 args

If we start at argv+2, then we can copy maximum argc-2 args to not go
into undefined territory.
---
 termux-api.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/termux-api.c b/termux-api.c
index 6d3673155c53..300c2a4e7d61 100644
--- a/termux-api.c
+++ b/termux-api.c
@@ -275,7 +275,7 @@ _Noreturn void exec_am_broadcast(int argc, char** argv,
     child_argv[14] = argv[1];
 
     // Copy the remaining arguments -2 for first binary and second api name:
-    memcpy(child_argv + extra_args, argv + 2, (argc-1) * sizeof(char*));
+    memcpy(child_argv + extra_args, argv + 2, (argc-2) * sizeof(char*));
 
     // End with NULL:
     child_argv[argc + extra_args - 1] = NULL;
-- 
2.49.0

